First pass at using RFC 2119 keywords

duglin opened the pull request:
Also:
- removed some editorial comments that aren't really appropriate for a spec
- added a tool to verify we upper-case our RFC2119 keywords

Signed-off-by: Doug Davis <dug@us.ibm.com>

carimura commented:
Overall LGTM. Agree with formatting, etc.

duglin commented:
@ultrasaurus not sure what you mean by the "formatting in the preamble".  I removed a chunk of text that I didn't think was appropriate for a spec, but the rest of it should be just removing of words that are reserved for RFC2119 usage (or fixing a few minor typos).

duglin commented on tools/verify-phrases.sh:
```diff
@@ -0,0 +1,180 @@
+#!/bin/bash
```
This script will check to make sure all uses of the RFC2119 keywords are upper-cased. This will help make sure that we don't include them by mistake.

ultrasaurus replied:
@duglin there are a lot of changes from top of file to ### namespace -- pulling those into a separate PR will make it easier for reviewer to focus on the things that could potentially change the meaning of the spec

duglin replied:
yes but a lot of them are related to removing rfc2119 keywords in places that we didn't mean to use them.

ultrasaurus replied:
True, though just thought the reviews might be easier in two parts that are logically separate, even though they are related.  Just a suggestion.

duglin commented on spec.md:
```diff
@@ -103,133 +84,162 @@ occurrence that tools and developers can rely upon to better handle the event.
   the event.
+* Constraints:
+  * OPTIONAL
+  * If present, MUST be a non-empty string
```
I was trying to keep things consistent with the current doc and save the harder discussions of changing the semantics for follow-on PRs. But let's see what other people think about making this a required field (I need to check too...)

duglin replied:
made it REQUIRED for now so we can try to resolve this PR on the next call. We can always revert it later if we want.

duglin commented on spec.md:
```diff
@@ -103,133 +84,162 @@ occurrence that tools and developers can rely upon to better handle the event.
     while open-events-version covers the context.
 
 ### open-events-version
 * Type: String
```
Seems reasonable.  Any objection to this one being REQUIRED?  @ac360 ?

duglin replied:
Done

ultrasaurus commented on spec.md:
```diff
@@ -103,133 +84,162 @@ occurrence that tools and developers can rely upon to better handle the event.
     this schema.
+  * TBD is it allowed to be empty if present?
 
 ## Context Attributes Backlog
```
Note: in today's meeting we decided to move these into bugs, so you might want to remove from this PR to avoid merge conflict.

duglin replied:
yep - done!

rabbah commented on spec.md:
```diff
@@ -1,6 +1,7 @@
 # CloudEvents - Verson 0.1
 
-CloudEvents is a vendor-neutral specification for event data.
```
note typo above Verson -> Version.

duglin replied:
Even though I didn't touch that line as part of this PR, I fixed it. thanks!

duglin commented on spec.md:
```diff
@@ -103,133 +84,162 @@ occurrence that tools and developers can rely upon to better handle the event.
 ### schema-url
-* Type: String
+* Type: URI per [RFC 3986](https://tools.ietf.org/html/rfc3986)
 * Description: A link to the schema which can be optionally specified by the
```
good catch - fixed

duglin commented on spec.md:
```diff
@@ -103,133 +84,162 @@ occurrence that tools and developers can rely upon to better handle the event.
+  * If present, MUST be a non-empty string
 
 ### resource
 * Type: Object
```
No strong opinion on this one yet so I'll make it REQUIRED. We can always change our minds later if needed.

duglin commented:
I addressed all comments.

One thing I was wondering about is how someone will deal with our REQUIRED properties. For example, what if the producer just doesn't have a "namespace" to associate with the event? Does this mean that they can ever be CloudEvents compliant? Or must they just generate a dummy value? I'm not making a formal suggestion yet, but I do kind of wonder if we should word things to be more like "this property is OPTIONAL, but if you do include this data then it MUST use this property name".  Then we still get interop w/o forcing people to inject dummy values that don't really provide any value.  As I said, no formal proposal, but I'd like to hear from event producers to know how they feel about our REQUIRED properties.

duglin commented:
The travis failure is due to some incorrectly cased keywords in the backlog section that will be removed in #39. So #39 needs to be merged first.

duglin commented:
Approved on Feb 1st call
